js/web/lib/wasm/jsep/webgpu/ops/skip-layer-norm.ts

```diff
@@ -124,35 +124,61 @@ const createSkipLayerNormProgramInfo =
               variables.push(outputVariable('input_skip_bias_sum', inputs[0].dataType, outputShape, components));
             }
             const dataType = tensorTypeToWsglStorageType(inputs[0].dataType);
+            const vecDataType = tensorTypeToWsglStorageType(DataType.float, components);
             return `
 
       ${shaderHelper.registerUniforms(uniformsArray).declareVariables(...variables)}
+      var<workgroup> sumShared : array<${vecDataType}, ${workgroupSize}>;
+      var<workgroup> sumSquaredShared : array<${vecDataType}, ${workgroupSize}>;
+
+      ${shaderHelper.mainStart([
+              workgroupSize, 1, 1
+            ])}
+        let ix = local_id.x;
+        let iy = global_id.x / ${workgroupSize};
 
-      ${shaderHelper.mainStart()}
-        ${shaderHelper.guardAgainstOutOfBoundsWorkgroupSizes('uniforms.output_size / uniforms.hidden_size')}
         let hidden_size_vectorized: u32 = uniforms.hidden_size / uniforms.components;
-        let offset = global_idx * hidden_size_vectorized;
-        var sum = ${fillVector('f32', components)};
-        var squareSum = ${fillVector('f32', components)};
-        for (var i: u32 = 0; i < hidden_size_vectorized; i++) {
+        var stride = hidden_size_vectorized / ${workgroupSize};
+        let offset = ix * stride + iy * hidden_size_vectorized;
+        let offset1d = stride * ix;
+        if (ix == ${workgroupSize - 1}) {
+          stride = hidden_size_vectorized - stride * ix;
+        }
+        for (var i: u32 = 0; i < stride; i++) {
           let skip_value = skip[offset + i];
-          let bias_value = ${hasBiasInput ? 'bias[i]' : dataType + '(0.0)'};
+          let bias_value = ${hasBiasInput ? 'bias[offset1d + i]' : dataType + '(0.0)'};
           let input_value = x[offset + i];
           let value = input_value + skip_value + bias_value;
           ${hasInputSkipBiasSumOutput ? 'input_skip_bias_sum[offset + i] = value;' : ''}
           output[offset + i] = value;
           let f32_value = ${castToF32(dataType, components, 'value')};
-          sum += f32_value;
-          squareSum += f32_value * f32_value;
+          sumShared[ix] += f32_value;
+          sumSquaredShared[ix] += f32_value * f32_value;
         }
+        workgroupBarrier();
+
+        var reduceSize : u32 = ${workgroupSize};
```

Comment:
in current shader code we use `snake_case` instead of `camelCase` for identifiers.

Reply:
fix